COLLABORATOR_GUIDE.md

```diff
@@ -263,6 +263,20 @@ multiple commits. Commit metadata and the reason for the revert should be
 appended. Commit message rules about line length and subsystem can be ignored.
 A Pull Request should be raised and approved like any other change.
 
+### Introducing New Modules
+
+Semver-minor commits that introduce new core modules should be treated with
+extra care.
+
+* First, the name of the new core module should not conflict with any existing
+  module in the ecosystem unless an agreement with the owner of those modules
+  is reached to transfer ownership.
+* PRs that introduce new core modules must be landed as semver-minor commits.
+* New core modules must be landed initially as `Experimental` and must go
+  through at least one release cycle before being moved out of experimental.
```

Comment:
The term "release cycle" is probably vague to a lot of people.

Reply:
So, yeah, putting my comments on the second and third bullet points together, maybe consolidate them into this single bullet point:

> * New core modules must be landed with a [Stability Index](./doc/api/documentation.md#stability-index) of `Experimental`. (Pull requests proposing new core modules must still be marked `semver-minor`.) New core modules must remain `Experimental` until a `semver-major` release.